modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/messaging/test/IndividualSegmentsCheckerTest.java

```diff
@@ -0,0 +1,293 @@
+									JSONUtil.putAll(
+										JSONUtil.put(
+											"id", "1234567"
+										).put(
+											"name", "Test segment"
+										)))
+							).put(
+								"page",
+								JSONUtil.put(
+									"number", 0
+								).put(
+									"size", 100
+								).put(
+									"totalElements", 1
+								).put(
+									"totalPages", 1
+								)
+							).put(
+								"total", 0
+							).toString()
+						).put(
+							"/api/1.0/individual-segments/1234567/individuals",
+							() -> JSONUtil.put(
+								"_embedded",
+								JSONUtil.put(
+									"individuals",
+									JSONUtil.putAll(
+										JSONUtil.put(
+											"dataSourceIndividualPKs",
+											JSONUtil.putAll(
+												JSONUtil.put(
+													"dataSourceId", "123456789"
+												).put(
+													"individualPKs",
+													JSONUtil.putAll(
+														_user.getUuid())
+												)))))
+							).put(
+								"page",
+								JSONUtil.put(
+									"number", 0
+								).put(
+									"size", 100
+								).put(
+									"totalElements", 1
+								).put(
+									"totalPages", 1
+								)
+							).put(
+								"total", 0
+							).toString()
+						).build()));
+
+			ReflectionTestUtil.invoke(
+				_individualSegmentsChecker, "checkIndividualSegments",
+				new Class<?>[0]);
+
+			List<SegmentsEntry> segmentsEntryList =
+				_segmentsEntryLocalService.getSegmentsEntries(
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+			Assert.assertEquals(
+				segmentsEntryList.toString(), 1, segmentsEntryList.size());
+
+			SegmentsEntry segmentsEntry = segmentsEntryList.get(0);
+
+			Assert.assertEquals(
+				"Test segment",
+				segmentsEntry.getName(LocaleUtil.getSiteDefault()));
+
+			List<SegmentsEntryRel> segmentsEntryRelList =
+				_segmentsEntryRelLocalService.getSegmentsEntryRels(
+					segmentsEntry.getSegmentsEntryId());
+
+			Assert.assertEquals(
+				segmentsEntryRelList.toString(), 1,
+				segmentsEntryRelList.size());
+
+			SegmentsEntryRel segmentsEntryRel = segmentsEntryRelList.get(0);
+
+			Assert.assertEquals(
+				_user.getUserId(), segmentsEntryRel.getClassPK());
+		}
+	}
+
+	private Object _getInstance(String className, Bundle bundle)
+		throws Exception {
+
+		Class<?> clazz = bundle.loadClass(className);
+
+		Constructor<?> constructor = clazz.getConstructor();
+
+		return constructor.newInstance();
+	}
+
+	private void _setIndividualSegmentsCheckerTest() throws Exception {
+		Bundle bundle1 = FrameworkUtil.getBundle(
+			IndividualSegmentsCheckerTest.class);
+
+		BundleContext bundleContext = bundle1.getBundleContext();
+
+		Bundle bundle2 = null;
+
+		for (Bundle curBundle : bundleContext.getBundles()) {
+			if (Objects.equals(
+					curBundle.getSymbolicName(),
+					"com.liferay.segments.asah.connector")) {
+
+				bundle2 = curBundle;
+
+				break;
+			}
+		}
+
+		Assert.assertNotNull(
+			"Unable to find com.liferay.segments.asah.connector bundle",
+			bundle2);
+
+		_individualSegmentsChecker = _getInstance(
```

Comment:
Why don't you inject directly the _individualSegmentsChecker?

Instead creating a new one, finding the IndividualSegmentsChecker that is already registered?